spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/metadata/MongoDbMetadataStore.java

```diff
@@ -0,0 +1,197 @@
+ * restarts.
+ *
+ * @author Senthil Arumugam, Samiraj Panneer Selvam
+ * @since 4.2
+ *
+ */
+public class MongoDbMetadataStore implements ConcurrentMetadataStore {
+
+	protected final Log logger = LogFactory.getLog(getClass());
+	private final MongoTemplate template;
+	private final String DEFAULT_COLLECTION_NAME = "metadatastore";
+	private final String collectionName;
+	private final String ID_FIELD = "_id";
+	private final String VALUE = "value";
+	
+	/**
+	 * 
+	 * Configures the MongoDbMetadataStore by provided {@link MongoTemplate} and
+	 * default collection name - {@link #DEFAULT_COLLECTION_NAME}.
+	 * 
+	 * @param template The mongodb template
+	 * @param collectionName the collection name where it persists the data
+	 */
+	public MongoDbMetadataStore(MongoTemplate template, String collectionName) {
+		this.template = template;
+		this.collectionName = (StringUtils.hasText(collectionName)) ? collectionName
+				: DEFAULT_COLLECTION_NAME;
+	}
+
+	/**
+	 * Configures the MongoDbMetadataStore by provided {@link MongoTemplate} and
+	 * default collection name - {@link #DEFAULT_COLLECTION_NAME}.
+	 * 
+	 * @param template The mongodb template
+	 */
+	public MongoDbMetadataStore(MongoTemplate template) {
+		this(template, null);
+	}
+
+	/**
+	 * Configures the MongoDbMetadataStore by provided {@link MongoDbFactory} and
+	 * default collection name - {@link #DEFAULT_COLLECTION_NAME}.
+	 * 
+	 * @param factory The mongodb factory
+	 */
+	public MongoDbMetadataStore(MongoDbFactory factory) {
+		this(factory, null);
+	}
+	
+	/**
+	 * Configures the MongoDbMetadataStore by provided {@link MongoDbFactory} and
+	 * collection name
+	 * 
+	 * @param collectionName the collection name where it persists the data
+	 * @param factory MongoDbFactory
+	 */
+	public MongoDbMetadataStore(MongoDbFactory factory, String collectionName) {
+		template = new MongoTemplate(factory);
+		this.collectionName = (StringUtils.hasText(collectionName)) ? collectionName
+				: DEFAULT_COLLECTION_NAME;
+	}
+	
+	
+	/**
+	 * Method put.
+	 * @param key String
+	 * @param value String
+	 * @see org.springframework.integration.metadata.MetadataStore#put(String, String)
+	 */
+	@Override
+	public void put(String key, String value) {
+		Assert.notNull(key, "'key' must not be null.");
+		Assert.notNull(value, "'value' must not be null.");
+		Map<String,String> fileInfo = new HashMap();
+		fileInfo.put(ID_FIELD, key);
+		fileInfo.put(VALUE, value);
+		template.save(fileInfo, collectionName);
+	}
+
+	/**
+	 * Method get.
+	 * @param key String
+	 * @return String
+	 * @see org.springframework.integration.metadata.MetadataStore#get(String)
+	 */
+	@Override
+	public String get(String key) {
+		Assert.notNull(key, "'key' must not be null.");
+		Query query = new Query(Criteria.where(ID_FIELD).is(key));
+		@SuppressWarnings("unchecked")
+		Map<String,String> result = template.findOne(query, Map.class, collectionName);
+		return result==null?null:result.get(VALUE);
```

Comment:
It isn't so easy to provide such a tool, hence we just follow with code style rules: https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Code-Style.
Some of them you can import to your IDE: 
https://github.com/spring-projects/spring-framework/tree/master/src/idea
https://github.com/spring-projects/spring-framework/tree/master/src/eclipse
But of course, there is no such an option to format your whitespaces vulnerabilities :smile: